source/fe/fe_enriched.cc

```diff
@@ -1048,6 +1049,451 @@ FE_Enriched<dim, spacedim>::InternalData::get_fe_output_object(
+    }
+
+
+
+    template <int dim, int spacedim>
+    void
+    make_fe_collection_from_colored_enrichments(
+      const unsigned int &                       num_colors,
+      const std::vector<std::set<unsigned int>> &fe_sets,
+      const std::vector<std::function<const Function<spacedim> *(
+        const typename Triangulation<dim, spacedim>::cell_iterator &)>>
+        &                              color_enrichments,
+      const FE_Q<dim, spacedim> &      fe_base,
+      const FE_Q<dim, spacedim> &      fe_enriched,
+      const FE_Nothing<dim, spacedim> &fe_nothing,
+      hp::FECollection<dim, spacedim> &fe_collection)
+    {
+      // define dummy function which is associated with FE_Nothing
+      using cell_function = std::function<const Function<spacedim> *(
+        const typename Triangulation<dim, spacedim>::cell_iterator &)>;
+      cell_function dummy_function;
+      dummy_function =
+        [=](const typename Triangulation<dim, spacedim>::cell_iterator &)
+        -> const Function<spacedim> * {
+        AssertThrow(false,
+                    ExcMessage("Called enrichment function for FE_Nothing"));
+        return nullptr;
+      };
+
+      using EnrichmentFunctions_2DVector =
+        std::vector<std::vector<std::function<const Function<spacedim> *(
+          const typename Triangulation<dim, spacedim>::cell_iterator &)>>>;
+
+      // loop through color sets and create FE_enriched element for each
+      // of them provided before calling this function, we have color
+      // enrichment function associated with each color.
+      for (unsigned int color_set_id = 0; color_set_id != fe_sets.size();
+           ++color_set_id)
+        {
+          std::vector<const FiniteElement<dim, spacedim> *> vec_fe_enriched(
+            num_colors, &fe_nothing);
+          EnrichmentFunctions_2DVector functions(num_colors, {dummy_function});
+
+          for (auto it = fe_sets[color_set_id].begin();
+               it != fe_sets[color_set_id].end();
+               ++it)
+            {
+              // Given a color id ( = *it), corresponding color enrichment
+              // function is at index id-1 because color_enrichments is
+              // is indexed from zero.
+              const unsigned int ind = *it - 1;
+
+              AssertIndexRange(ind, vec_fe_enriched.size());
+              AssertIndexRange(ind, functions.size());
+              AssertIndexRange(ind, color_enrichments.size());
+
+              // Assume an active predicate colors {1,2} for a cell.
+              // We then need to create a vector of FE enriched elements
+              // with vec_fe_enriched[0] = vec_fe_enriched[1] = &fe_enriched
+              // which can later be associated with enrichment functions.
+              vec_fe_enriched[ind] = &fe_enriched;
+
+              // color_set_id'th color function is (color_set_id-1)
+              // element of color wise enrichments
+              functions[ind].assign(1, color_enrichments[ind]);
```

Comment:
i think 
```
funcitons[ind][0] = color_enrichments[ind];
```
is more readable as you already have the right size of single element with default being `dummy_function`